mina.netty/src/main/java/org/jboss/netty/channel/socket/nio/NioClientDatagramChannelFactory.java

```diff
@@ -89,73 +87,40 @@
  * <p>
  * Multicast is not supported.  Please use {@link OioDatagramChannelFactory}
  * instead.
- *
- * @apiviz.landmark
  */
-public class NioDatagramChannelFactory implements DatagramChannelFactory {
+public class NioClientDatagramChannelFactory implements DatagramChannelFactory {
 
     private final NioDatagramPipelineSink sink;
-    private final NioChildDatagramPipelineSink childSink;
-    private final WorkerPool<NioDatagramWorker> workerPool;
-    private final WorkerPool<NioWorker> childPool;
+    private final WorkerPool<NioWorker> workerPool;
     private final InternetProtocolFamily family;
     private boolean releasePool;
 
-    /**
-     * Create a new {@link NioDatagramChannelFactory} with a {@link Executors#newCachedThreadPool()}
-     * and without preferred {@link InternetProtocolFamily}.  Please note that the {@link InternetProtocolFamily}
-     * of the channel will be platform (and possibly configuration) dependent and therefore
-     * unspecified.  Use {@link #NioDatagramChannelFactory(InternetProtocolFamily)} if unsure.
-     *
-     * See {@link #NioDatagramChannelFactory(Executor)}
-     */
-    public NioDatagramChannelFactory() {
-        this((InternetProtocolFamily) null);
-    }
-
-    /**
-     * Create a new {@link NioDatagramChannelFactory} with a {@link Executors#newCachedThreadPool()}.
-     *
-     * See {@link #NioDatagramChannelFactory(Executor)}
-     */
-    public NioDatagramChannelFactory(InternetProtocolFamily family) {
-        workerPool = new NioDatagramWorkerPool(Executors.newCachedThreadPool(), SelectorUtil.DEFAULT_IO_THREADS);
-        childPool = new NioWorkerPool(Executors.newCachedThreadPool(), SelectorUtil.DEFAULT_IO_THREADS);
-        this.family = family;
-        sink = new NioDatagramPipelineSink(workerPool);
-        childSink = new NioChildDatagramPipelineSink(childPool);
+    public NioClientDatagramChannelFactory(WorkerPool<NioWorker> workerPool) {
+        this.workerPool = workerPool;
+        this.family = null;
+        sink = new NioDatagramPipelineSink();
         releasePool = true;
     }
 
-    public DatagramChannel newChannel(final ChannelPipeline pipeline) {
-        return new NioDatagramChannel(this, pipeline, sink, sink.nextWorker(), family);
-    }
-
-    // mina.netty change -  adding this to create child datagram channels
-    public NioChildDatagramChannel newChildChannel(Channel parent, final ChannelPipeline pipeline) {
-        return new NioChildDatagramChannel(parent, this, pipeline, childSink, childSink.nextWorker(), family);
+    public DatagramChannel newChannel(ChannelPipeline pipeline) {
+        return new NioDatagramChannel(this, pipeline, sink, workerPool.nextWorker(), family);
```

Comment:
Keep the `final` modifier on the `pipeline` method parameter since this is not a required change.